src/liballoc/boxed.rs

```diff
@@ -145,23 +145,23 @@ pub struct IntermediateBox<T: ?Sized> {
            issue = "27779")]
 unsafe impl<T> Place<T> for IntermediateBox<T> {
     fn pointer(&mut self) -> *mut T {
-        self.ptr as *mut T
+        self.ptr.cast().as_ptr()
     }
 }
 
 unsafe fn finalize<T>(b: IntermediateBox<T>) -> Box<T> {
-    let p = b.ptr as *mut T;
+    let p = b.ptr.cast().as_ptr();
     mem::forget(b);
     Box::from_raw(p)
 }
 
 fn make_place<T>() -> IntermediateBox<T> {
     let layout = Layout::new::<T>();
 
-    let p = if layout.size() == 0 {
-        mem::align_of::<T>() as *mut u8
-    } else {
-        unsafe {
+    let p = unsafe {
+        if layout.size() == 0 {
+            NonNull::new_unchecked(mem::align_of::<T>() as *mut u8)
```

Comment:
Maybe `NonNull::<T>::dangling().cast()`?

Reply:
Indeed.

Reply:
Note that #48333 removes this code.

Reply:
But there's another in raw_vec.rs.